libs/domains/service-helm/feature/src/lib/networking-port-setting-modal/networking-port-setting-modal.tsx

```diff
@@ -76,71 +91,118 @@ export function NetworkingPortSettingModal({ port, onClose, onSubmit }: Networki
         }
       >
         <div className="flex flex-col gap-5">
-          <Controller
-            name="service_name"
-            defaultValue=""
-            control={control}
-            rules={{
-              required: 'Please enter service name.',
-            }}
-            render={({ field, fieldState: { error } }) => (
-              <InputText
-                name={field.name}
-                onChange={(e: FormEvent<HTMLInputElement>) => {
-                  const name = `p${watchInternalPort}-${e.currentTarget.value}`
-                  setValue('name', name.slice(0, namePatternRules.maxLength.value))
-                  field.onChange(e)
+          {isLoading ? (
+            <div className="flex justify-center">
+              <LoaderSpinner />
+            </div>
+          ) : availableServiceNames && availableServiceNames?.length > 0 ? (
+            <Controller
+              name="service_name"
+              defaultValue=""
+              control={control}
+              rules={{
+                required: 'Please enter service name.',
+              }}
+              render={({ field, fieldState: { error } }) => (
+                <InputSelect
+                  label="Service name"
+                  value={field.value}
+                  options={availableServiceNames}
+                  error={error?.message}
+                  onChange={(e) => {
+                    const name = `p${watchInternalPort}-${e}`
+                    setValue('name', name.slice(0, namePatternRules.maxLength.value))
+                    field.onChange(e)
+                  }}
+                  portal
+                />
+              )}
+            />
+          ) : (
+            <Controller
+              name="service_name"
+              defaultValue=""
+              control={control}
+              rules={{
+                required: 'Please enter service name.',
+              }}
+              render={({ field, fieldState: { error } }) => (
+                <InputText
+                  name={field.name}
+                  onChange={(e: FormEvent<HTMLInputElement>) => {
+                    const name = `p${watchInternalPort}-${e.currentTarget.value}`
+                    setValue('name', name.slice(0, namePatternRules.maxLength.value))
+                    field.onChange(e)
+                  }}
+                  value={field.value}
+                  label="Service name"
+                  error={error?.message}
+                />
+              )}
+            />
+          )}
+          {watchServiceName ? (
+            isLoading ? (
```

Comment:
I wonder if the best is to have an global wrapper `isLoading` and hide other fields if it's not loaded, wdyt? It avoid space gap when `isLoading` is finally false

Reply:
I'm not really sensitive to that.
I've made the change
https://github.com/Qovery/console/pull/1654/commits/a37dfefcad1b8c178aefaff7c4fea957cf935259
tell me if you like it better.